pkg/k8s/watchers/watcher.go

```diff
@@ -640,12 +657,32 @@ func genCartesianProduct(
 							Protocol: fePort.Protocol,
 							Port:     fePort.Port,
 						},
+						Scope: loadbalancer.ScopeExternal,
 					},
 					ID: loadbalancer.ID(0),
 				},
 				Backends: besValues,
 				Type:     svcType,
 			})
+
+		if svcSize > len(ports) {
```

Comment:
please add a comment to this condition, I would even create a new variable on top to be more explicit.
e.g. `isTrafficPolicyLocal := svcSize > len(ports)`